Bugfix related to USGS stations

SorooshMani-NOAA opened the pull request:
Fixes #76

SorooshMani-NOAA commented:
@brey it looks like both are now working fine: https://mybinder.org/v2/gh/SorooshMani-NOAA/searvey/bugfix/examples
The pull is ready to be merged. Note that we do need a release because of the bugfix related to `pandas`

brey commented:
> I'm trying to test the branch on binder. It takes some time for binder Jupyter to come up, do you have the same issue with binder? Is it related to the number of package dependencies?

Yes. There are a number of potential reasons/measures. See [here](https://discourse.jupyter.org/t/how-to-reduce-mybinder-org-repository-startup-time/4956) for more info.

pmav99 commented on searvey/usgs.py:
```diff
@@ -126,8 +126,8 @@ def normalize_usgs_stations(df: pd.DataFrame) -> gpd.GeoDataFrame:
-    df.end_date = pd.to_datetime(df.end_date, errors="coerce")
-    df.begin_date = pd.to_datetime(df.begin_date, errors="coerce")
+    df.end_date = pd.to_datetime(df.end_date, errors="coerce", format="mixed")
+    df.begin_date = pd.to_datetime(df.begin_date, errors="coerce", format="mixed")
```
I think this is not the proper fix. 

The `format` argument does exist in pandas 1.X but `mixed` is not a special value in 1.X. `format="mixed"` was only added in 2.0. Consequently, even though `pd.to_datetime()` does not raise an Exception in pandas 1.X it always returns `NaT`.

For example this will raise an Assertion in 2.X but not in 1.X
``` python
import pandas as pd
assert pd.to_datetime("2001-09-23", errors="coerce", format="mixed") is pd.NaT
```

I did spend too much time on this, but I can't find a cleaner solution than checking the pandas version with `importlib.metadata.version()`

https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.to_datetime.html
https://pandas.pydata.org/pandas-docs/version/2.0/reference/api/pandas.to_datetime.html

pmav99 replied:
I pushed a fix. Not sure if it is the best solution though. If anyone things of something better, please let me know

SorooshMani-NOAA replied:
@pmav99 thanks for noticing this. I think what you did makes the most sense.

brey replied:
I am not sure what exactly we need here, but if it to test whether the time stamp is not available or `pd.Nat` this should work for both versions:

```python
if pd.isna(pd.to_datetime("NaN")):
     do something
```

pmav99 replied:
It's actually the opposite. We don't want to detect `NaT` but to avoid generating `NaT`s in the first place. In order to do so we need an extra argument in `pd.to_datetime()` but the value of this argument is not valid in 1.X. That's why we need to detect the pandas version and conditionally pass this argument.

pmav99 commented:
I added a small test and rebased in order to keep the git history clean. Test coverage is decreasing, but I am not sure it is worth it to setup CI to run with both pandas 1 and 2.

SorooshMani-NOAA commented:
> I added a small test and squashed/rebased in order to keep the git history clean. Test coverage is decreasing, but I am not sure it is worth it to setup CI to run with both pandas 1 and 2.

@pmav99 Thanks, I don't think it's worth it.
